Make build_timestamp generation less platform specific.
This also slightly changes the format of the timezone
offset.

$ date -Iseconds
2017-06-21T13:33:15+02:00

$ date +%Y-%m-%dT%H:%M:%S%z
2017-06-21T13:33:24+0200

--- Makefile.orig	2017-01-20 19:41:51.000000000 +0100
+++ Makefile	2017-02-08 13:11:51.099665000 +0100
@@ -666,7 +666,7 @@
 $(BUILDDIR)/timestamp.c: FORCE
 	@mkdir -p $(dir $@)
 	@echo '#include "build.h"' > $@
-	@echo 'const char* build_timestamp = "'`date -Iseconds`'";' >> $@
+	@echo 'const char* build_timestamp = "'`date +%Y-%m-%dT%H:%M:%S%z`'";' >> $@
 
 $(BUILDDIR)/timestamp.o: $(BUILDDIR)/timestamp.c
 	$(pCC) -c -o $@ $<
